src/app/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import PwResetIcon from "../Icons/PwResetIcon";
+import RequestIcon from "../Icons/RequestIcon";
+import SearchIcon from "../Icons/SearchIcon";
+import UserIcon from "../Icons/UserIcon";
+import styles from "./IconButton.module.css";
+
+type IconButtonProps = {
+  onClick: () => void;
+  variant: "login" | "signUp" | "pwReset" | "search" | "newSearch" | "request";
+};
+
+const variants = {
+  login: { icon: <UserIcon />, title: "Login" },
+  signUp: { icon: <UserIcon />, title: "Sign up" },
+  pwReset: { icon: <PwResetIcon />, title: "Reset password" },
+  search: { icon: <SearchIcon />, title: "Search" },
+  newSearch: { icon: <SearchIcon />, title: "New search" },
+  request: { icon: <RequestIcon />, title: "Send request" },
+};
```

Comment:
I would pass the variant and the text separately

Reply:
Thanks for your review! Today, I decided to pass icon and title separately (without defining variants) to be flexible with my button in the future. In case I need an additional icon for example.